src/Core/src/Handlers/Window/WindowHandler.iOS.cs

```diff
@@ -82,12 +102,48 @@ public static void MapMenuBar(IWindowHandler handler, IWindow view)
 		public static void MapRequestDisplayDensity(IWindowHandler handler, IWindow window, object? args)
 		{
 			if (args is DisplayDensityRequest request)
+			{
 				request.SetResult(handler.PlatformView.GetDisplayDensity());
+			}
 		}
 
 		void UpdateVirtualViewFrame(UIWindow window)
 		{
 			VirtualView.FrameChanged(window.Bounds.ToRectangle());
 		}
+
+		class WindowProxy
+		{
+			WeakReference<IWindow>? _virtualView;
+
+			IWindow? VirtualView => _virtualView is not null && _virtualView.TryGetTarget(out var v) ? v : null;
+			IDisposable? _effectiveGeometryObserver;
+
+			public void Connect(IWindow virtualView, UIWindow platformView)
+			{
+				_virtualView = new(virtualView);
+				_effectiveGeometryObserver = platformView.WindowScene?.AddObserver("effectiveGeometry", NSKeyValueObservingOptions.OldNew, HandleEffectiveGeometryObserved);
```

Comment:
@MartyIX while you are adding logging, can you add a comment that links to the apple docs:
https://developer.apple.com/documentation/uikit/uiwindowscene/effectivegeometry?language=objc#Discussion

> This property is key-value observing (KVO) compliant. Observing [effectiveGeometry](https://developer.apple.com/documentation/uikit/uiwindowscene/effectivegeometry?language=objc) is the recommended way to receive notifications of changes to the window scene’s geometry. These changes can occur because of user interaction or as a result of the system resolving a geometry request.

I know some properties are not KVO compliant, so we basically ran in fear from all of this. But Apple says this is OK so we can do it. But future maui people are going to be scared. At least I was.